app/reconciler/createChangeSetFromCompareStructures.ts

```diff
@@ -0,0 +1,103 @@
+import { isEqual } from 'lodash';
+import logger from '~/lib/logger.server';
+
+import { DnsRecordType } from '@prisma/client';
+
+import type { Change } from '@aws-sdk/client-route-53';
+import type { ReconcilerCompareStructure } from './ReconcilerTypes';
+
+interface CompareStructures {
+  dbStructure: ReconcilerCompareStructure;
+  route53Structure: ReconcilerCompareStructure;
+}
+
+export const createRemovedChangeSetFromCompareStructures = ({
+  dbStructure,
+  route53Structure,
+}: CompareStructures) => {
+  const toChange: Change[] = [];
+
+  /**
+   * If something is present in route53 but is completely missing in db,
+   * then we have to delete it
+   */
+  Object.keys(route53Structure).forEach((fqdn) => {
```

Comment:
I tried asking ChatGPT to improve readability of this code, interesting to see what it did:

```ts
import { isEqual } from 'lodash';
import logger from '~/lib/logger.server';

import { DnsRecordType } from '@prisma/client';

import type { Change } from '@aws-sdk/client-route-53';
import type { ReconcilerCompareStructure } from './ReconcilerTypes';

interface CompareStructures {
  dbStructure: ReconcilerCompareStructure;
  route53Structure: ReconcilerCompareStructure;
}

export function createDeleteChangeSet({ dbStructure, route53Structure }: CompareStructures) {
  const toChange: Change[] = [];

  for (const [fqdn, route53Types] of Object.entries(route53Structure)) {
    for (const [type, route53Value] of Object.entries(route53Types)) {
      const dbValue = dbStructure[fqdn]?.[type as DnsRecordType];

      if (!dbValue) {
        toChange.push({
          Action: 'DELETE',
          ResourceRecordSet: {
            MultiValueAnswer: route53Value.length > 1,
            Name: fqdn,
            Type: type,
            ResourceRecords: route53Value.map((Value) => ({ Value })),
            TTL: 60 * 5,
          },
        });
      }
    }
  }

  return toChange;
}

export function createUpsertChangeSet({ dbStructure, route53Structure }: CompareStructures) {
  const toChange: Change[] = [];

  for (const [fqdn, dbTypes] of Object.entries(dbStructure)) {
    for (const [type, dbValue] of Object.entries(dbTypes)) {
      const route53Value = route53Structure[fqdn]?.[type as DnsRecordType];

      if (!route53Value || !isEqual(dbValue?.sort(), route53Value?.sort())) {
        const isMultivalue =
          dbValue.length > 1 &&
          [DnsRecordType.A, DnsRecordType.AAAA, DnsRecordType.TXT].includes(type as DnsRecordType);

        if (!isMultivalue) {
          logger.error(
            'Error creating DNS changeset: only A, AAAA, and TXT records can be multivalue. Ignoring records.',
            { fqdn, type }
          );
          continue;
        }

        toChange.push({
          Action: 'UPSERT',
          ResourceRecordSet: {
            MultiValueAnswer: isMultivalue,
            Name: fqdn,
            Type: type,
            ResourceRecords: dbValue.map((Value) => ({ Value })),
            TTL: 60 * 5,
          },
        });
      }
    }
  }

  return toChange;
}
```

Reply:
Well, it wrote code, but it's not the same functionality, like this is completely incorrect:
```js
const isMultivalue =
          dbValue.length > 1 &&
          [DnsRecordType.A, DnsRecordType.AAAA, DnsRecordType.TXT].includes(type as DnsRecordType);

        if (!isMultivalue) {
          logger.error(
            'Error creating DNS changeset: only A, AAAA, and TXT records can be multivalue. Ignoring records.',
            { fqdn, type }
          );
          continue;
        }
```

Reply:
Totally, I was just interested in its style.  Not suggesting you use it.  Fascinating to see what it can and can't do.